core/src/main/java/jenkins/widgets/HistoryPageEntry.java

```diff
@@ -59,14 +59,7 @@ protected static long getEntryId(@Nonnull Object entry) {
             return ((Queue.Item) entry).getId();
         } else if (entry instanceof Run) {
             Run run = (Run) entry;
-            long queueId = run.getQueueId();
-            if (queueId == Run.QUEUE_ID_UNKNOWN) {
-                // Backward compatibility. This is a run from before the Queue.Item IDs
-                // were mapped onto their resulting Run instance.
-                return (Integer.MIN_VALUE + run.getNumber());
-            } else {
-                return queueId;
-            }
+            return (Integer.MIN_VALUE + run.getNumber());
```

Comment:
This should work, but I think it should actually use `Long.MIN_VALUE`.

So, more generally, I think this could result in funky pagination behaviour as items transition from queue items to runs, where the page boundary lies at the point where items transitioned and the user clicks at the right time - they'll appear on both pages.